### ISSUE TITLE
Adding paypal processor

### ISSUE THREAD

**Issue by SunyataZero** [peoples-fund/peoples-fund#18]
Main advantage (for us) over stripe is that _paypal does not require the user to have a credit card_. See details in the pasted text below

---

https://www.quora.com/Why-should-I-switch-our-websites-to-Stripe-from-Paypal

Julia Grace, Head of Engineering at Tindie, Runner, Python, JavaScript
Written Sep 29, 2013
As someone who has used (and implemented) both Stripe and PayPal on production sites to accept a non-trivial number of payments, I would rephrase this question to be:

```
Should I add Stripe to our sites that use PayPal (and vice versa)? 
```

Tindie ran PayPal alone for well over a year before we added Stripe (but did not take away PayPal). HumbleBundle.com uses 5 payment options: PayPal, Stripe, Google Checkout, Amazon Payments and Bitcoin.

If you are initially adding a payment processor, I suggest you ask yourself:

```
What demographic am I targeting and what are their preferences when making online payments? 
```

You will need to balance the answer to this question with the implementation and maintenance of whatever solution(s) you choose.

If you live in the US and have a US customer demographic, accepting credit cards appears to be the obvious answer. But what if you have a global customer base that includes countries where people don't have, use or trust credit cards? They may not trust PayPal either, but then you have a bigger problem.

Something else to note is that while the US developer community has a strong, vocal distrust for PayPal, if your customer demographic is not developers then they might hate PayPal a lot less than you think.

tl;dr version:

```
If you only care about price, you are in the US and have US only customers with credit cards, Stripe is easier and faster to implement and off-the-shelf price is the same as PayPal. Use Stripe.
If you only care about price, you are in the US and you have a significant number of international customers with credit cards, the Stripe off-the-shelf rate is cheaper than PayPal. Use Stripe.
If your bank is not in one of the countries Stripe supports (Stripe: Global) then use PayPal. Use PayPal.
If your customer demographic is largely outside the US and doesn't have, use or trust credit cards, then Stripe can't help you. Use PayPal.
If you have a mixed customer base of US and International customers, Use both.
```

Longer version (here are my personal experiences with both):

Pricing

```
Stripe is currently at 2.9% + 30 cents in the US (Pricing - United States) but on other pages they say "Do you offer discounted pricing at scale? If you plan to process more than 10,000 payments this year, drop us a line…" (https://stripe.com/global). I have heard people negotiate a lesser rate, but that rate is likely based on volume.
PayPal is also currently at 2.9% + 30 cents in the US (PayPal Merchant Fees) but the extra 1% that Christina mentions for international payments is painful.  In my experience with PayPal once you hit a certain threshold they will automatically lower your rate. I don't know if Stripe does this.
```

Are you a non-profit?

```
Stripe does not give you a discount (Does Stripe offer a fee discount for non-profit organizations?). 
PayPal apparently cuts you a deal if you are a registered non-profit (PayPal Merchant Fees).
```

Time to integrate

```
Stripe is much easier to integrate with and in part that is because as Praveen mentioned, Stripe just processes credit cards. Stripe documentation does an excellent job with getting you from zero to accepting credit cards very quickly. If you want to do more complex things (for example, store "customers", allow those customers to update their credit cards, etc.) I have found the docs lacking there, in part I hypothesize because less people use those features.
Paypal is much more complex, has many different APIs with different permissions (classic APIs such as Adaptive Payments, Adaptive Accounts, Merchant APIs, etc. and now the new OAuth2 based APIs, etc.) Learning how PayPal IPNs (Instant Payment Notification ) work is tricky. Is the complexity an excuse for confusing documentation? Of course not, but it's important to note because this is not a strict apples to apples comparison since time to integrate depends on what you're trying to do.
```

Inconsistencies between production and development

```
Stripe: I have yet to see a production/development inconsistency (I'm not saying it doesn't happen, I'm just saying I have never seen it, nor have I heard of this happening).
PayPal: I have found a handful of times when differences exist between  production and development (PayPal calls this the sandbox). For example, I once saw a case where in production you are required to pass variables that you can get away with not passing in development. As an engineer this deeply scares me.
```

Documentation

```
Stripe documentation is quite good and as I mentioned earlier can get you up and running quickly.
PayPal documentation can be confusing in part because of the sheer amount of docs they have available. I have found the challenge is not what the docs say (although they can be confusing) but rather which doc is the right one for what you're trying to do. 

Note: Something very distressing about PayPal is that the docs are occasionally wrong. Sometimes you follow the instructions, you get an error back you didn't expect and you change your code, and all is well. Sometimes your code just won't work and no error is returned, then you resort to StackOverflow, searching PayPal forums, etc. I have called PayPal business support and gotten wrong answers from them. I have asked PayPal developers in person why their API has no way to query for the fee on a specific transaction (when you don't have a PRO account) and they stare blankly at me and say "You mean our APIs don't do that?" (I then go through an example with them, and they are shocked that I'm right).
```

User experience

```
Stripe has the "Stripe Checkout" (Checkout) feature where Stripe builds and styles the credit card form for you (you drop in some JavaScript and it takes care of this). The credit card form shows up as a lightbox and the user never leaves the site. You can also style the form as well to look like your site. Love it.
PayPal requires us to redirect to PayPal to accept the Payment, then redirect back to our site. Etsy (product) and Humble Bundle do this in their PayPal implementations as well. It's horrible and the last thing I want as an eCommerce platform is to redirect users away from my site. It decreases conversion, users think it's shady, etc.
```

Chances you'll get shutdown

```
Stripe: I personally do not know of Stripe shutting down an account, but one has to believe this must happen since fraud is a real concern.
PayPal: Nearly everyone I know who has reached some level of "success" (meaning they are processing a non-trivial number and amount of payments) has had their PayPal account temporarily frozen. My understanding is the purpose of this is to investigate if the account is fraudulent. This brings your entire business to a screeching halt while you get things sorted out. Daniel Greenfeld documents what happened in the open source community: We are not using PayPal.
```

Disclaimer: I know Praveen Alavilli (he's helped me debug several PayPal problems) and I also have close personal friends at Stripe.
